Source/DafnyCore/AST/Function.cs

```diff
@@ -84,6 +84,7 @@ public TailStatus
   public readonly List<TypeParameter> TypeArgs;
   public readonly List<Formal> Formals;
   public readonly Formal Result;
+  public PreType ResultPreType;
```

Comment:
Why is `ResultType` readonly, but `ResultPreType` isn't?

Reply:
`Result` is determined by the parser, whereas `ResultPreType` happens at the beginning of pre-type resolution.